spring-batch-bigquery/src/main/java/org/springframework/batch/extensions/item/google/bigquery/BigQueryItemWriter.java

```diff
@@ -0,0 +1,437 @@
+                    "Dataset should be configured properly"
+            );
+        }
+
+        createDataset();
+    }
+
+    private boolean isCsv() {
+        return FormatOptions.csv().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isJson() {
+        return FormatOptions.json().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isAvro() {
+        return FormatOptions.avro().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isParquet() {
+        return FormatOptions.parquet().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isOrc() {
+        return FormatOptions.orc().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isBigtable() {
+        return FormatOptions.bigtable().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isGoogleSheets() {
+        return FormatOptions.googleSheets().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean isDatastore() {
+        return FormatOptions.datastoreBackup().getType().equals(this.writeChannelConfig.getFormat());
+    }
+
+    private boolean tableHasDefinedSchema(Table table) {
+        return Optional
+                .ofNullable(table)
+                .map(Table::getDefinition)
+                .map(TableDefinition.class::cast)
+                .map(TableDefinition::getSchema)
+                .isPresent();
+    }
+
+    private void createDataset() {
+        if (Objects.nonNull(this.datasetInfo)) {
+            Dataset foundDataset = this.bigQuery.getDataset(this.writeChannelConfig.getDestinationTable().getDataset());
+
+            if (Objects.isNull(foundDataset)) {
+                this.bigQuery.create(this.datasetInfo);
+            }
+        }
+    }
+
+    /**
+     * It is expected that for {@link BigQueryItemWriter#isParquet()} and {@link BigQueryItemWriter#isOrc()} you use
+     * {@link Path} because there is no convenient way to write records to {@link OutputStream}.
+     * @deprecated not supported right now
+     */
+    @Deprecated
+    private Stream<byte[]> getHadoopPathByteArrayStream(List<? extends T> items) throws IOException {
+        Stream<byte[]> result = Stream.empty();
+
+        T firstElement = Iterables.getFirst(items, null);
+        Assert.notNull(firstElement, "Collection is empty");
+        Class<?> classType = firstElement.getClass();
+
+        if (classType.isAssignableFrom(Path.class)) {
+            List<URI> uris = items.stream()
+                    .map(Path.class::cast)
+                    .map(Path::toUri)
+                    .collect(Collectors.toList());
+
+            return getFileBasedByteArrayStream(uris);
+        }
+
+        return result;
+    }
+
+    @Deprecated
```

Comment:
Same comment as above with regard to deprecating private methods.

Reply:
Done